lsp/src/saros/lsp/extensions/server/contact/ContactService.java

```diff
@@ -0,0 +1,141 @@
+package saros.lsp.extensions.server.contact;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.function.BiPredicate;
+import saros.lsp.extensions.client.ISarosLanguageClient;
+import saros.lsp.extensions.server.SarosResponse;
+import saros.lsp.extensions.server.SarosResultResponse;
+import saros.lsp.extensions.server.contact.dto.AddInput;
+import saros.lsp.extensions.server.contact.dto.ContactDto;
+import saros.lsp.extensions.server.contact.dto.RemoveInput;
+import saros.lsp.extensions.server.contact.dto.RenameInput;
+import saros.lsp.ui.UIInteractionManager;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.IContactsUpdate;
+import saros.net.xmpp.contact.XMPPContact;
+import saros.net.xmpp.contact.XMPPContactsService;
+
+public class ContactService implements IContactService, IContactsUpdate {
+
+  private XMPPContactsService contactsService;
+  private ISarosLanguageClient client;
+  private UIInteractionManager interactionManager;
+
+  public ContactService(
+      XMPPContactsService contactsService,
+      ISarosLanguageClient client,
+      UIInteractionManager interactionManager) {
+    this.contactsService = contactsService;
+    this.client = client;
+    this.interactionManager = interactionManager;
+
+    this.contactsService.addListener(this);
+  }
+
+  @Override
+  public CompletableFuture<SarosResponse> add(AddInput input) {
+
+    CompletableFuture<SarosResponse> c = new CompletableFuture<SarosResponse>();
+
+    Executors.newCachedThreadPool()
```

Comment:
Have you read the API what this method does ?! Do you think it is that smart to spawn a thread pool for each method invocation of add ?